server/lib/nl/fulfillment/utils.py

```diff
@@ -270,3 +269,11 @@ def get_places_as_string(places: List[str]) -> str:
     return places[0]
   else:
     return ', '.join(places[0:len(places) - 1]) + f' and {places[-1]}'
+
+
+def get_max_ans_places(places: List[Place], uttr: Utterance) -> List[Place]:
```

Comment:
For later, I wonder if we can simplify logic branching based on some first order flags, and have them in one giant file or function...

That function / object itself can even be plotted to reason about the decision tree...

Reply:
Yeah, I think we can collect all the flags in one place.  I wonder if the decision tree might need to be built by propagating state in a context rather than putting all the decision tree code in one place (it is very hard to navigate that and would make things not very modular).  But yes, def something to improve on.